packages/rpc-methods/src/restricted/manageAccounts.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  SubjectType,
+  PermissionType,
+  RestrictedMethodOptions,
+  ValidPermissionSpecification,
+  PermissionSpecificationBuilder,
+} from '@metamask/permission-controller';
+import { Json, JsonStruct, NonEmptyArray } from '@metamask/utils';
+import { ethErrors } from 'eth-rpc-errors';
+import { assert, string, tuple } from 'superstruct';
+
+export const methodName = 'snap_manageAccounts';
+
+export type ManageAccountsMethodHooks = {
+  /**
+   * Gets the snap keyring implementation.
+   */
+  getSnapKeyring: (snapOrigin: string) => Promise<{
+    handleKeyringSnapMessage: (snapId: string, message: any) => Promise<Json>;
```

Comment:
Can the message be typed here? If it's a JSON-RPC request, we can use `JsonRpcRequest`, and otherwise maybe just `Json`?

Reply:
Updated in [de277d2](https://github.com/MetaMask/snaps/pull/1290/commits/de277d26a7810d3336f9fb5c63f868bb001cd9a8)